arithmetization/src/main/java/net/consensys/linea/zktracer/module/hub/precompiles/ModexpMetadata.java

```diff
@@ -37,14 +35,14 @@ public class ModexpMetadata {
   public static final int BASE_MIN_OFFSET = 0x60;
 
   private final Bytes callData;
-  private final EWord rawLeadingWord;
   @Setter private Bytes rawResult;
 
-  public ModexpMetadata(final Hub hub, final Bytes callData) {
+  public ModexpMetadata(final Bytes callData) {
     this.callData = callData;
-    int exponentOffsetInCallData = BASE_MIN_OFFSET + bbsInt();
-    this.rawLeadingWord =
-        EWord.of(hub.messageFrame().shadowReadMemory(exponentOffsetInCallData, WORD_SIZE));
+  }
+
+  public EWord rawLeadingWord() {
+    return EWord.of(rightPadSlice(callData, BASE_MIN_OFFSET + bbsInt(), WORD_SIZE));
```

Comment:
LGTM

Reply:
Why remove the check that was present ?
```java
checkArgument(loadRawLeadingWord());
```
Did it ever blow up ?

Reply:
the function was never used.